### DISCUSSION TITLE
Welcome to Betfair Discussions!

### DISCUSSION THREAD

**Discussion by KelvinVail** [KelvinVail/Betfair#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by BetOpz** [comment]
Hi Kelvin, brilliant work! Just curious why you decided to leave name out of the market def? My own fault for not reading properly but spent ages trying to get name as the betfair docs say it is available. I would say it is extremely important for trading apps and having to call REST for the names is far less efficient. Do you have any intentions of releasing a version with it available?

**Reply by KelvinVail** [reply]
Thanks for reaching out. Could you give me a little more info? Are you talking about the MarketDefinition returned on the stream? I'll double and test but I don't believe market name is available there. If you're calling the rest endpoint for market catalogue, it should be available there.

**Comment by BetOpz** [comment]
Hi Kelvin,
 
Thanks for taking the time to get back to me.
 
I was talking about selection names from the marketdefinition - some 
docs seem to suggest they were intentionally left out of the library 
even though they are part of the marketdefinition?
I ended up getting names via rest and caching them to be used inside the 
stream push that I'm building. I just assumed selection names would be 
available from marketdefinition as betfair docs kept saying they are 
available.
After admitting defeat, I re-read the docs for your library and 
realised...........my own fault for skimming!
 
Kind Regards.
 
Ray.
------ Original Message ------
From: ***@***.***
To: ***@***.*** Cc: ***@***.***; 
***@***.***
Sent: Sunday, September 21st 2025, 10:28
Subject: Re: [KelvinVail/Betfair] Welcome to Betfair Discussions! 
(Discussion #4)
Thanks for reaching out. Could you give me a little more info? Are you 
talking about the MarketDefinition returned on the stream? I'll double 
and test but I don't believe market name is available there. If you're 
calling the rest endpoint for market catalogue, it should be available 
there. -- Reply to this email directly or view it on GitHub: 
https://github.com/KelvinVail/Betfair/discussions/4#discussioncomment-14467824

**Reply by KelvinVail** [reply]
No problem at all. Getting selection names is a bit confusing. I went through the same thing 😁
Betfair only provides the names via the rest API. I'm guessing Betfair are intentionally keeping stream messages as small as possible and only including details that might change.

If you have any other feedback on the library, do let me know. I love hearing from users.
